dev/validate_bom__test.py

```diff
@@ -430,33 +430,30 @@ def __init__(self, deployer):
         'echo-worker': 8089
     }
 
-    # Map of services with provided hosts and credentials.
-    outbound_service_configs = {}
-    if self.options.test_gate_service_host:
-      gate_service_config = {
-        'host': self.options.test_gate_service_host,
-      }
-      bearer_auth_token = self.__gate_service_bearer_auth_token_or_none()
-      if bearer_auth_token:
-        gate_service_config['bearer_auth_token'] = bearer_auth_token
-      if self.options.test_gate_iap_credentials:
-        gate_service_config['service_account_email'] = get_service_account_email(self.options.test_gate_iap_credentials)
-      outbound_service_configs['gate'] = gate_service_config
-
-    self.__outbound_service_configs = outbound_service_configs
-
-  def __gate_service_bearer_auth_token_or_none(self):
-    iap_credentials_file = self.options.test_gate_iap_credentials
-    iap_client_id = self.options.test_gate_iap_client_id
-
-    if iap_credentials_file and not iap_client_id:
-      raise_and_log_error(
-          ConfigError('Cannot set --test_gate_iap_credentials field without setting --test_gate_iap_client_id'))
-    if not iap_credentials_file and iap_client_id:
-      raise_and_log_error(
-          ConfigError('Cannot set --test_gate_iap_client_id field without setting --test_gate_iap_credentials'))
-    if iap_credentials_file and iap_client_id:
-      return generate_auth_token(iap_credentials_file, iap_client_id)
+    # Map of services with provided endpoints and credentials.
+    self.__outbound_service_configs = {}
+    self.__add_gate_service_config(self.__outbound_service_configs)
+
+  def __add_gate_service_config(self, configs):
+    if not self.options.test_gate_service_endpoint:
+      return
+    gate_service_config = {
```

Comment:
all the local variable gate_ prefix here are superfluous because the method itself is scoped to gate.
Removing them will make it easier to read.